ui/src/app/Pages/company/company.component.ts

```diff
@@ -1,31 +1,90 @@
 import { Component, OnInit } from '@angular/core';
 import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { PageEvent } from '@angular/material/paginator';
+import { BehaviorSubject } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
+
+import { Company } from 'src/app/domain/company/models/company';
+import { CompanyRepository } from '../../domain/company/company.repository';
 
 @Component({
   selector: 'app-company',
   templateUrl: './company.component.html',
   styleUrls: ['./company.component.sass']
 })
 export class CompanyComponent implements OnInit {
-  companyForm!:FormGroup;
-  displayedColumns: string[] = ['Question', 'description'];
-  isVisible:boolean = false;
-  constructor(private formBuilder:FormBuilder) { }
+  companyForm!: FormGroup;
+  displayedColumns: string[] = ['id', 'name', 'description', 'update', 'delete'];
+  isVisible: boolean = false;
+  isAppear: boolean = false;
```

Comment:
try to make names more specific
make Difference between isVisible and isAppear each one for what